@dnhatn I think the scroll that is failing is actually the previous scroll from the same test, which starts on line 10: `java.lang.AssertionError: Failure at [upgraded_cluster/10_basic:10]: expected [2xx] status code...`. This is the same test failure that the PR targets -- the call is attempting to continue a scroll that started before the upgrade was complete, but that scroll has expired. I think that bumping the scroll keep-alive time on the initial (pre-upgrade) request is the right way to address this.

There is probably some confusion because I referred to a scroll 'timeout' as opposed to a keep-alive time. I'll correct this in the PR title/ description.

--- a/x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/mixed_cluster/10_basic.yml
+++ b/x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/mixed_cluster/10_basic.yml
@@ -32,7 +32,7 @@
         rest_total_hits_as_int: true
         index: upgraded_scroll
         size: 1
-        scroll: 8m
+        scroll: 10m
         sort: foo
         body:
           query:

